instance/tests/test_gandi.py

```diff
@@ -37,69 +38,88 @@ class GandiTestCase(TestCase):
     """
     def setUp(self):
         super().setUp()
+        self.api = gandi.GandiAPI(client=FakeGandiClient())
 
-        with patch('xmlrpc.client.ServerProxy'):
-            self.api = gandi.GandiAPI()
-            self.api.client.domain.info.return_value = {
-                'fqdn': 'test.com',
-                'zone_id': 9900,
-                # .... full API response contains more fields, but we don't need them in this test.
-            }
+    def populate_cache(self):
+        """
+        Populate the zone ID cache of the Gandi API and reset the mocks.
+        """
+        self.api._populate_zone_id_cache()
+        self.api.client.domain.reset_mock()
 
     def assert_set_dns_record_calls(self, attempts=1):
         """
         Verify Gandi API calls for setting a DNS record value.
         """
-        domain_info_call = call.domain.info('TEST_GANDI_API_KEY', 'test.com')
-        create_new_zone_version_call = call.domain.zone.version.new('TEST_GANDI_API_KEY', 9900)
-        delete_old_record_call = call.domain.zone.record.delete(
-            'TEST_GANDI_API_KEY', 9900, 'new_zone_version',
+        create_new_zone_version_call = call.zone.version.new('TEST_GANDI_API_KEY', 9900)
+        delete_old_record_call = call.zone.record.delete(
+            'TEST_GANDI_API_KEY', 9900, 1,
             {'type': ['A', 'CNAME'], 'name': 'sub.domain'}
         )
-        create_new_record_call = call.domain.zone.record.add(
-            'TEST_GANDI_API_KEY', 9900, 'new_zone_version',
+        create_new_record_call = call.zone.record.add(
+            'TEST_GANDI_API_KEY', 9900, 1,
             {'value': '192.168.99.99', 'ttl': 1200, 'type': 'A', 'name': 'sub.domain'}
         )
-        set_new_zone_version_call = call.domain.zone.version.set('TEST_GANDI_API_KEY', 9900, 'new_zone_version')
+        set_new_zone_version_call = call.zone.version.set('TEST_GANDI_API_KEY', 9900, 1)
 
         self.assertEqual(
-            self.api.client.mock_calls,
-            [domain_info_call] +
+            self.api.client.domain.mock_calls,
             [create_new_zone_version_call] * attempts +
             [delete_old_record_call, create_new_record_call, set_new_zone_version_call]
         )
 
+    def test_populate_zone_id_cache(self):
+        """
+        Test that populating the zone ID cache results in the expected API calls.
+        """
+        self.api._populate_zone_id_cache()
```

Comment:
Your test coverage is awesome 😄   Only [gandi.py#L81](https://github.com/open-craft/opencraft/blob/a19ed4fecd2733c900407d5e5995385a61bcf296/instance/gandi.py#L81) is missing from the unit tests.

You could bump `gandi.py`'s coverage from 96% to 98% by simply changing this line to:

    self.api.get_zone_id('test.com')